File: Task-03/prime.cpp
#include <iostream>
#include <vector>
#include <cmath>

bool isPrime(int num) {
    if (num < 2) return false;
    for (int i = 2; i * i <= num; i++) {
        if (num % i == 0) return false;
    }
    return true;
}

int main() {
    int n;
    std::cout << "Enter a number: ";
    std::cin >> n;
    std::vector<int> primes;

    for (int i = 2; i <= n; i++) {
        if (isPrime(i)) {
            primes.push_back(i);
        }
    }

    std::cout << "Prime numbers up to " << n << " are: ";
    for (int prime : primes) {
        std::cout << prime << " ";
    }
    std::cout << std::endl;

    return 0;
}

File: Task-03/prime.hs
isPrime :: Int -> Bool
isPrime num
    | num < 2 = False
    | otherwise = all (\i -> num `mod` i /= 0) [2..isqrt]
    where isqrt = floor (sqrt (fromIntegral num))

findPrimes :: Int -> [Int]
findPrimes n = [x | x <- [2..n], isPrime x]

main :: IO ()
main = do
    putStrLn "Enter a number: "
    n <- readLn
    let primes = findPrimes n
    putStrLn $ "Prime numbers up to " ++ show n ++ " are: " ++ show primes

